balancer-js/src/modules/pools/factory/linear/linear.factory.ts

```diff
@@ -0,0 +1,291 @@
+import { LogDescription } from '@ethersproject/abi';
+import { BytesLike } from '@ethersproject/bytes';
+import { Contract } from '@ethersproject/contracts';
+import { JsonRpcProvider, TransactionReceipt } from '@ethersproject/providers';
+
+import { BalancerError, BalancerErrorCode } from '@/balancerErrors';
+import {
+  AaveLinearPool__factory,
+  AaveLinearPoolFactory__factory,
+  ERC4626LinearPool__factory,
+  ERC4626LinearPoolFactory__factory,
+  EulerLinearPool__factory,
+  EulerLinearPoolFactory__factory,
+  GearboxLinearPool__factory,
+  GearboxLinearPoolFactory__factory,
+  YearnLinearPool__factory,
+  YearnLinearPoolFactory__factory,
+} from '@/contracts';
+import { AaveLinearPoolInterface } from '@/contracts/AaveLinearPool';
+import { AaveLinearPoolFactoryInterface } from '@/contracts/AaveLinearPoolFactory';
+import { ERC4626LinearPoolInterface } from '@/contracts/ERC4626LinearPool';
+import { ERC4626LinearPoolFactoryInterface } from '@/contracts/ERC4626LinearPoolFactory';
+import { EulerLinearPoolInterface } from '@/contracts/EulerLinearPool';
+import { EulerLinearPoolFactoryInterface } from '@/contracts/EulerLinearPoolFactory';
+import { GearboxLinearPoolInterface } from '@/contracts/GearboxLinearPool';
+import { GearboxLinearPoolFactoryInterface } from '@/contracts/GearboxLinearPoolFactory';
+import { YearnLinearPoolInterface } from '@/contracts/YearnLinearPool';
+import { YearnLinearPoolFactoryInterface } from '@/contracts/YearnLinearPoolFactory';
+import { ContractInstances } from '@/modules/contracts/contracts.module';
+import { PoolFactory } from '@/modules/pools/factory/pool-factory';
+import {
+  InitJoinPoolAttributes,
+  LinearCreatePoolParameters,
+  ProtocolId,
+} from '@/modules/pools/factory/types';
+import { PoolType } from '@/types';
+import { findEventInReceiptLogs } from '@/lib/utils';
+
+type LinearPoolFactoryInterface =
+  | AaveLinearPoolFactoryInterface
+  | ERC4626LinearPoolFactoryInterface
+  | EulerLinearPoolFactoryInterface
+  | GearboxLinearPoolFactoryInterface
+  | YearnLinearPoolFactoryInterface;
+
+type LinearPoolFactoryInterfaceWithoutYearn =
+  | AaveLinearPoolFactoryInterface
+  | ERC4626LinearPoolFactoryInterface
+  | EulerLinearPoolFactoryInterface;
+
+type LinearPoolParamsToEncode = [
+  string,
+  string,
+  string,
+  string,
+  string,
+  string,
+  string,
+  string
+];
+
+type YearnLinearPoolParamsToEncode = [
```

Comment:
What do you think about turning these types into objects, so we're able to read each parameter more easily?
Then, you could use something like Object.values to fill out encodeCreateFunctionData params.